Lib/_collections_abc.py

```diff
@@ -1094,9 +1084,11 @@ def reverse(self):
     def extend(self, values):
         'S.extend(iterable) -- extend sequence by appending elements from the iterable'
         if values is self:
-            values = list(values)
-        for v in values:
-            self.append(v)
+            for i in range(len(values)):
+                self.append(values[i])
```

Comment:
I'm not comfortable changing this.  We can't know in advance whether the underling code will respond well to mutating while iterating over itself. It is safer to have a copy of the data.  Also, we don't know that indexing is a fast operation (i.e. for a linked list implementation indexing is O(n)).  Lastly, the C code for list_extend() does exactly what the current code is doing (making a copy). We want to stay faithful to that implementation.

Reply:
Okay.